messaging/README.md

```diff
@@ -1,22 +1,38 @@
-Firebase Cloud Messaging Quickstart
-===================================
+# Firebase Cloud Messaging Quickstart
 
 The Firebase Cloud Messaging quickstart demonstrates how to:
+
 - Request permission to send app notifications to the user.
 - Receive FCM messages using the Firebase Cloud Messaging JavaScript SDK.
 
-Introduction
-------------
+## Introduction
+
+### Sending
+
+There are two types FCM messages you can send:
+
+- `notification message`: messages that are displayed by the SDK on developers behalf.
+- `data message`: messages that contains developer-specified key value pairs.
+
+It is recommended to read more about the differences and use cases in [_About FCM messages_](https://firebase.google.com/docs/cloud-messaging/concept-options).
 
-[Read more about Firebase Cloud Messaging](https://firebase.google.com/docs/cloud-messaging/)
+### Receiving
```

Comment:
This content might be more helpful in context, down in the "App focus" section.

I'd suggest linking to the existing version in the docs (https://firebase.google.com/docs/cloud-messaging/js/receive) rather than recreating it here, though.  Make sense?